src/blobs.cpp

```diff
@@ -66,10 +66,36 @@ namespace rhost {
             }
         }
 
-        bool to_blob(SEXP sexp, rhost::util::blob& blob) {
+        bool to_blob(SEXP sexp, std::vector<char>& blob) {
             bool result = false;
             rhost::util::errors_to_exceptions([&] {result = to_blob_internal(sexp, blob); });
             return result;
         }
+
+        void append_from_file(blob& blob, const char* path) {
```

Comment:
Yep. This should work currently, provided that they don't use any Unicode letters in the name that cannot be mapped into their default codepage. If they do, we'll break (but, again, it's not really a new issue).

For future refactoring, to mitigate this, we should start using `std::filesystem` types everywhere throughout the code. As I recall, those are supposed to be platform-aware, and use the internal representation that can represent any path (e.g. UCS2 on Windows, UTF-8 on Mac, native locale on Linux etc). I'll file a separate issue.